### ISSUE TITLE
WebSocket `concatenateFragmentedFrames` is a no-op

### ISSUE THREAD

**Issue by kamilkloch** [softwaremill/tapir#3339]
Current implementation of `optionallyConcatenateFrames` returns the original stream:

https://github.com/softwaremill/tapir/blob/fdfa3da9a7d96b49625412a3aebed3ffe4e87988/server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala#L71

**Comment by adamw** [comment]
Do you see where the problem is? I guess it would be high time to add some tests for the `pipeToBody` specifically

**Comment by kamilkloch** [comment]
Accumutor is always `None` - lines https://github.com/softwaremill/tapir/blob/fdfa3da9a7d96b49625412a3aebed3ffe4e87988/server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala#L76 through https://github.com/softwaremill/tapir/blob/fdfa3da9a7d96b49625412a3aebed3ffe4e87988/server/http4s-server/src/main/scala/sttp/tapir/server/http4s/Http4sWebSockets.scala#L79 are never invoked.

**Comment by kamilkloch** [comment]
Question, if such functionality is actually needed. http4s concatenates web socket frames by default, not sure about the other servers.

**Comment by kamilkloch** [comment]
But, astonishingly, nobody is complaining about non-concatenating the frames by tapir interpreter for other servers.

**Comment by adamw** [comment]
Yes, maybe these are unpopular servers, or people don't send fragmented frames that often. I don't know really :)
